tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Kafka/KafkaConsumerConstructorIntegration.cs

```diff
@@ -51,6 +52,23 @@ public class KafkaConsumerConstructorIntegration
                 }
             }
 
+            var originalHandler = consumer.OffsetsCommittedHandler;
+            var dataStreams = Tracer.Instance.TracerManager.DataStreamsManager;
+            consumer.OffsetsCommittedHandler = (consumer, result) =>
+            {
+                originalHandler?.Invoke(consumer, result);
```

Comment:
Not sure if this follows best practices, I'm open to optimization suggestions :)

Reply:
I think this is a perfect case to use https://github.com/DataDog/dd-trace-dotnet/pull/4613 . You need to change the `OffsetsCommittedHandler` type in the DuckType proxy definition with `Delegate` and then just wrap it and assign the new delegate.